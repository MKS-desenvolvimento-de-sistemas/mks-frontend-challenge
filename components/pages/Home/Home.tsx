import Head from "next/head";
import Header from "@/components/Header/Header";
import Products from "@/components/Products/Products";
import { useCallback, useEffect, useState } from "react";
import mks, { Product } from "../../../integrations/mks";
import ShoppingCartDrawer from "@/components/ShoppingCartDrawer/ShoppingCartDrawer";
import Footer from "../../../Footer/Footer";
import styled from "styled-components";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
`;

export default function Home() {
  const [products, setProducts] = useState<Product[]>([]);
  const [loading, setLoading] = useState(true);
  const [drawerVisible, setDrawerVisible] = useState(false);

  const openDrawer = () => {
    setDrawerVisible(true);
  };

  const closeDrawer = () => {
    setDrawerVisible(false);
  };

  const fetchProducts = useCallback(async () => {
    setLoading(true);

    try {
      const fetchedProducts = await mks.fetchProducts({
        page: 1,
        rows: 50,
        sortBy: "id",
        orderBy: "ASC",
      });

      if (fetchedProducts) {
        setProducts(fetchedProducts);
      }
    } finally {
      setLoading(false);
    }
  }, []);

  useEffect(() => {
    fetchProducts().catch(() => {
      console.error("Falha ao carregar produtos");
    });
  }, [fetchProducts]);

  return (
    <Container>
      <Head>
        <title>MKS Sistemas | E-commerce</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Header onOpenCart={openDrawer} />

      <main>
        <Products loading={loading} products={products} />
      </main>

      <ShoppingCartDrawer visible={drawerVisible} onClose={closeDrawer} />
      <Footer />
    </Container>
  );
}
