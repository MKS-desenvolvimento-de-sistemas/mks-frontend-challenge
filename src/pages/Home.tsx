import { useEffect, useState } from "react";
import styled from "styled-components";
import ControlPanel from "../ControlPanel";
import { getProducts } from "../Service/Api";
import Card from "../Card";
import Skeleton from "../components/Skeleton";
import { ICartItem } from "./Cart";

const HomeStyles = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  /* gap: 1rem; */
  flex-direction: column;
  & .cards {
    padding-top: 3rem;
    padding-bottom: 3rem;
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    gap: 1rem;
    justify-content: center;
    flex-wrap: wrap;
  }
`;

const Home = () => {
  const [count, setCount] = useState(0);
  const [listaProdutos, setListaProdutos] = useState<any>();
  const [isLoading, setIsLoading] = useState(true);

  const listarProdutos = async (page: number, rows: number) => {
    const produtos = await getProducts(page, rows);
    setIsLoading(false);
    setListaProdutos(produtos);
  };

  useEffect(() => {
    listarProdutos(1, 8);
  }, []);

  return (
    <>
      <HomeStyles>
        {isLoading && (
          <div
            style={{
              display: "flex",
              gap: "1rem",
              width: "100%",
              justifyContent: "center",
              alignItems: "center",
              paddingTop: "3.2rem",
            }}
          >
            <Skeleton height="312px" width="232px" />
            <Skeleton height="312px" width="232px" />
            <Skeleton height="312px" width="232px" />
            <Skeleton height="312px" width="232px" />
            <Skeleton height="312px" width="232px" />
          </div>
        )}
        {/* <ControlPanel
          listaProdutos={listaProdutos}
          listarProdutos={listarProdutos}
          setListaProdutos={setListaProdutos}
        /> */}
        <div className="cards">
          {!!listaProdutos &&
            listaProdutos.products.map((produto) => (
              <Card key={produto.id} produto={produto}></Card>
            ))}
        </div>
      </HomeStyles>
    </>
  );
};

export default Home;
