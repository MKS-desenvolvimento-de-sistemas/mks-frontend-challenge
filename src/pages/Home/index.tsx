import React from 'react';
import Header from '../../components/Header';
import List from '../../components/List';
import Cart from '../../components/Cart';

type Props = Record<string, unknown>;

const Home = (props: Props) => (
	<>
		<Cart />
	</>
);

export default Home;
