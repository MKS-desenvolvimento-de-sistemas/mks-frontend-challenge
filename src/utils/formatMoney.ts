const formatMoney = (price: string) => {
  return parseFloat(price).toLocaleString('pt-BR', {
    style: 'currency',
    currency: 'BRL',
    minimumFractionDigits: 0,
    maximumFractionDigits: 0,
  });
};

export default formatMoney;