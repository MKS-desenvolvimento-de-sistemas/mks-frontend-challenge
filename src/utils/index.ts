import { IProductCart } from "../types/api";

export const getQuantityInCart = (cart: IProductCart[]) => {
  if (cart.length === 0) return 0;
  const quantities = cart.map((product) => product.quantity);
  const quantityTotal = quantities.reduce(
    (acc, currentValue) => acc + currentValue
  );

  return quantityTotal;
};

export const getPriceInCart = (cart: IProductCart[]) => {
  if (cart.length === 0) return 0;
  const prices = cart.map(
    (product) => Number(product.price) * product.quantity
  );
  const priceTotal = prices.reduce((acc, currentValue) => acc + currentValue);

  return priceTotal;
};
