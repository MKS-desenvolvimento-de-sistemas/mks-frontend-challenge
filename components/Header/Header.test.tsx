import { render } from "@testing-library/react";
import Home from "@/pages/index";
import { withTheme } from "../../test/utils";

it("Header should match snapshot", () => {
  const { container } = render(withTheme(<Home />));
  expect(container).toMatchSnapshot();
});
