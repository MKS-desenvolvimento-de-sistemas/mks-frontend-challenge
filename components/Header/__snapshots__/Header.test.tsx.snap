// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <header
    class="sc-dkzDqf eNeVyn"
  >
    <div>
      <h1
        class="sc-bczRLJ bzdgXU"
      >
        MKS
      </h1>
      <h2
        class="sc-gsnTZi fojcSU"
      >
        Sistemas
      </h2>
    </div>
  </header>
</div>
`;
